COLLABORATOR_GUIDE.md

```diff
@@ -279,6 +245,150 @@ export default { component: NameOfComponent } as Meta;
+`next.dynamic.mjs` is responsible for getting a full list of the source pages (`pages/en`) and identifying which pages have been translated.
+
+Non-translated pages will have their Localized contexts and translated React message-bags (`react-intl`) but the content will be the same as the source page (English).
+Whereas localized pages will have localized context and content.
+
+This custom solution is also able to decide what paths should be compiled during runtime.
+This is a combination of rules defined on `next.constants.mjs` and `[...path].tsx`.
+
+The `[...path].tsx` file ultimately utilizes the `theme.tsx` file as its layout source.
+This setup enables the loading of the Layout Provider and MDX Provider, which in turn, encapsulate and manage the rendering of any child content or components. This includes both content and components provided by the Layout Provider and the transformed MDX content originating from the `.md(x)` source page.
+
+#### What is `next.data.mjs`?
+
+This file is responsible for loading, fetching and generating build-time required information such as Node.js Release data, Blog Posts Metadata (for pagination and indexation), RSS feeds, etc.
+
+#### What is `site.json`?
+
+This file is used for defining Website Metadata, such as which RSS feeds should be generated, Social Media Information and other Metadata used during the Website build-time.
+We use a JSON format to ease collaboration.
+
+#### What is `next.locales.mjs` and why not use Next.js built-in i18n?
+
+Next.js built-in i18n is great, but it is not flexible enough for our needs. We need to be able to generate a full list of supported locales, and we need to be able to generate a full list of supported pages for each locale. We also need to be able to generate a full list of supported pages for each locale, and we need to be able to generate a full list of supported pages for each locale.
+
+It is also important to mention that we use a subfolder approach, instead of extension-based (`filename.language.md(x)`) for our content.
+Which is trivially incompatible with Next.js built-in i18n.
+
+We chose this approach to maintain familiarity with the content-structure as much as possible with the previous Node.js Website and to ensure maintainability (not depending on Next.js's i18n feature).
+
+#### What is `next.rewrites.mjs`?
+
+This file is responsible for defining the rewrite rules for the Website.
+It is used for defining redirects and other rewrite rules. (Such as Internal Redirects and External ones).
+
+The actual rewrites are defined within `redirects.json`. The file format is simple and separate external redirects with internal rewrites.
+
+- Rewrites do not change the original URL but render the content of the rewritten URL. The reritten path must be valid and exist on the Website.
+- Redirects change the original URL to the new one. The URLs do not necessarily need to be external, but they in general should be.
```

Comment:
The last phrase is not clear to me: 

> but they in general should be.

should be what? Could you please elaborate

Reply:
"The URLs do not necessarily need to be external, but they in general should be.", meaning the redirect URLs should generally be external ones. It's implicit context 🤔

Reply:
After re-reading this phrase several times I think I got it now, probably worth rephrasing e.g. `In general the redirect URLs should be external but it is not necessary`

Reply:
I think both have the same meaning. At least for me it's easy to understand. Wdyt @mikeesto

Reply:
I understood the initial copy but I can appreciate there may be some confusion. Perhaps we can clarify with something like this:

```md
The `redirects.json` file specifies two types of operations: rewrites and redirects.

- Rewrites: These operations do not alter the original URL but instead render the content from the rewritten URL. It's important to note that the rewritten URL path must be valid and exist on the website.

- Redirects: Redirect operations, on the other hand, change the original URL to a new one. While these new URLs can be either internal (within the website) or external (leading to a different domain), it is more common for redirects to point externally. 
```